src/traces/scattergl/convert.js

```diff
@@ -279,18 +281,24 @@ proto.updateFast = function(options) {
         yy = y[i];
 
         // check for isNaN is faster but doesn't skip over nulls
-        if(!isNumeric(yy)) continue;
-        if(!isNumeric(xx) && !(xx instanceof Date)) continue;
+        fastType = isNumeric(xx) || xx instanceof Date;
 
-        idToIndex[pId++] = i;
+        if(isNumeric(yy) && (fastType || Lib.isDateTime(xx))) {
 
-        positions[ptr++] = xx;
-        positions[ptr++] = yy;
+            if(!fastType) {
+                xx = Lib.dateTime2ms(xx);
+            }
```

Comment:
@etpinard thanks for testing and making a decision over whether this approach is fast enough or not. The [current code in master](https://github.com/plotly/plotly.js/blob/master/src/traces/scattergl/convert.js#L282) already incurred the expense of two `isNumeric` calls per loop execution and admittedly I haven't attempted to make it faster than that. I agree that we can take this `Date` rework as an opportunity to speed up the loop a bit. For now, it can only be done probabilistically (for future options, I added a few lines of comment), as you said. In my previous commit I didn't shoot for such speedup, in part because it would be based on a heuristic, and in part because, if we want to increase speed, we might be able to do other things:

- if we want to speed up `plotly.js` time, then permitting a typed array input (= no array copy), or at least letting the user specify the value types can speed things up
- if we want to speed it up when called via Python, then either the Python interface could directly convert the date strings into JavaScript `Date` objects (or better, directly populate a typed array), or, the Python -> JS WebSockets interface can directly deserialize into a JS typed array (this of course assumes that Python sends epoch milliseconds).

Back to present time: the fast path can be provided if all values can be assumed to be number (or `Date`) rather than a majority, so I ended up not tampering with the `autotype` logic (it could be more DRY but wanted to keep loop bodies simple, and direct, for speed).